sway/commands/bar/bindsym.c

```diff
@@ -1,11 +1,69 @@
+#define _XOPEN_SOURCE 500
 #include <stdlib.h>
 #include <string.h>
+#include <strings.h>
 #include "sway/commands.h"
 #include "sway/config.h"
 #include "list.h"
 #include "log.h"
 #include "stringop.h"
 
 struct cmd_results *bar_cmd_bindsym(int argc, char **argv) {
-	return cmd_results_new(CMD_FAILURE, "bindsym", "TODO"); // TODO
+	struct cmd_results *error = NULL;
+	if ((error = checkarg(argc, "bar bindsym", EXPECTED_MORE_THAN, 1))) {
+		return error;
+	}
+	if (!config->current_bar) {
+		return cmd_results_new(CMD_FAILURE, "bar bindsym", "No bar defined.");
+	}
+
+	struct bar_binding *binding = calloc(1, sizeof(struct bar_binding));
+	if (!binding) {
+		return cmd_results_new(CMD_FAILURE, "bar bindsym",
+				"Unable to allocate bar binding");
+	}
+
+	binding->release = false;
+	if (strcmp("--release", argv[0]) == 0) {
+		binding->release = true;
+		argv++;
+		argc--;
+	}
+
+	binding->button = 0;
+	if (strncasecmp(argv[0], "button", strlen("button")) == 0 &&
+			strlen(argv[0]) == strlen("button0")) {
+		binding->button = argv[0][strlen("button")] - '1' + 1;
+	}
+	if (binding->button == 0) {
```

Comment:
Should there be a bounds check here? Like `> 0 && <= 9`? I'm not sure if it's possible to do something funky like `buttonA` which probably shouldn't be allowed.